core/src/main/java/io/grpc/internal/Channelz.java

```diff
@@ -450,21 +454,124 @@ public ChannelStats build() {
   }
 
   public static final class Security {
-    // TODO(zpencer): fill this in
+    @Nullable
+    public final Tls tls;
+    @Nullable
+    public final OtherSecurity other;
+
+    private Security(Tls tls, OtherSecurity other) {
+      Preconditions.checkState(
+          tls != null ^ other != null,
+          "one of tls or other should be set");
+      this.tls = tls;
+      this.other = other;
+    }
+
+    public static Security withTls(Tls tls) {
+      return new Security(tls, /*other=*/ null);
+    }
+
+    public static Security withOtherSecurity(OtherSecurity other) {
+      return new Security(/*tls=*/ null, other);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (!(o instanceof Security)) {
+        return false;
+      }
+      Security that = (Security) o;
+      return Objects.equal(tls, that.tls) && Objects.equal(other, that.other);
+    }
+
+    @Override
+    public int hashCode() {
+      // only tls field is immutable
+      return Objects.hashCode(tls);
+    }
+  }
+
+  public static final class OtherSecurity {
+    public final String name;
+    @Nullable
+    public final Object security;
+
+    public OtherSecurity(String name, @Nullable Object security) {
+      this.name = Preconditions.checkNotNull(name);
+      this.security = security;
+    }
+  }
+
+  @Immutable
+  public static final class Tls {
+    public final String cipherSuiteStandardName;
+    @Nullable public final String localCert;
+    @Nullable public final String remoteCert;
+
+    /**
+     * Creates an instance.
+     */
+    public Tls(
+        String cipherSuiteStandardName,
+        @Nullable String localCert,
+        @Nullable String remoteCert) {
+      this.cipherSuiteStandardName = cipherSuiteStandardName;
+      this.localCert = localCert;
+      this.remoteCert = remoteCert;
+    }
+
+    /**
+     * Creates an instance.
+     */
+    public static Tls fromSslSession(SSLSession session) {
+      String cipherSuiteStandardName = session.getCipherSuite();
+      String localCert = null;
+      String remoteCert = null;
+      Certificate[] localCerts = session.getLocalCertificates();
+      if (localCerts != null) {
+        localCert = localCerts[0].toString();
+      }
+      try {
+        Certificate[] peerCerts = session.getPeerCertificates();
+        if (peerCerts != null) {
+          remoteCert = peerCerts[0].toString();
```

Comment:
That's correct, I can add a comment since this is non obvious. According to the javadoc for `getPeerCertificates`:
  an ordered array of peer certificates, with the peer's own certificate first followed by any certificate authorities.